### ISSUE TITLE
`parse_html_for_line` doubles up already existing newlines

### ISSUE THREAD

**Issue by BenjaminRi** [trishume/syntect#318]
Suppose you use `SyntaxSet::load_defaults_nonewlines` and pass every line of your code string without line breaks to [`ClassedHTMLGenerator::parse_html_for_line`](https://docs.rs/syntect/4.4.0/syntect/html/struct.ClassedHTMLGenerator.html#method.parse_html_for_line). Newlines are preserved because `parse_html_for_line` automatically appends one for each line. So you end up with the same number of newlines as the original string. However, because the [documentation](https://docs.rs/syntect/4.4.0/syntect/parsing/struct.SyntaxSet.html#method.load_defaults_nonewlines) explicitly states that `load_defaults_nonewlines` is ["hacky" and "may not work" in some cases](https://docs.rs/syntect/4.4.0/syntect/parsing/struct.SyntaxSetBuilder.html#method.add_from_folder) and since I have those newlines anyway, I would like to use `load_defaults_newlines`.

Okay, imagine the same scenario with `SyntaxSet::load_defaults_newlines`. You supply every line with a line break as the last character to `parse_html_for_line`. Now, what happens is that newlines are doubled up. The problem is that `parse_html_for_line` automatically appends a newline to the already existing newline at the end of the line. Therefore, it is extremely inconvenient to use properly since I then need to run through (and copy) the entire finalized html string to filter out those doubled newlines.

The following sample code illustrates my problem. Notice how every line break is duplicated in the html output:

```rust
use syntect::html::{ClassedHTMLGenerator, ClassStyle};
use syntect::parsing::SyntaxSet;

fn main() -> Result<(), std::io::Error> {
	let ss = SyntaxSet::load_defaults_newlines();
	
	let code_cpp = "/* C++ source */
#include <iostream>
int main() {
	std::cout << \"Hello World!\" << std::endl;
}";

	let sr_cpp = ss.find_syntax_by_extension("cpp").unwrap();
	let mut rs_html_generator = ClassedHTMLGenerator::new_with_class_style(&sr_cpp, &ss, ClassStyle::Spaced);
	for line in code_cpp.lines() {
		//This line has no newline, that's why we add one for demonstration purposes:
		let line = line.to_owned() + "\n";
		println!("Cpp {:?}", &line);
		rs_html_generator.parse_html_for_line(&line);
	}
	let html_cpp = rs_html_generator.finalize();

	println!("{}", html_cpp);
	
	Ok(())
}
```

Output:

```
Cpp "/* C++ source */\n"
Cpp "#include <iostream>\n"
Cpp "int main() {\n"
Cpp "    std::cout << \"Hello World!\" << std::endl;\n"
Cpp "}\n"
```
```html
<span class="source c++"><span class="comment block c"><span class="punctuation definition comment c">/*</span> C++ source <span class="punctuation definition comment c">*/</span></span>

<span class="meta preprocessor include c++"><span class="keyword control import include c++">#include</span> <span class="string quoted other lt-gt include c++"><span class="punctuation definition string begin c++">&lt;</span>iostream<span class="punctuation definition string end c++">&gt;</span></span>
</span>
<span class="storage type c">int</span> <span class="meta function c++"><span class="entity name function c++">main</span></span><span class="meta function parameters c++"><span class="meta group c++"><span class="punctuation section group begin c++">(</span></span></span><span class="meta function parameters c++"><span class="meta group c++"><span class="punctuation section group end c++">)</span></span></span><span class="meta function c++"> </span><span class="meta function c++"><span class="meta block c++"><span class="punctuation section block begin c++">{</span></span></span><span class="meta function c++"><span class="meta block c++">

    std<span class="punctuation accessor c++">::</span>cout <span class="keyword operator arithmetic c">&lt;&lt;</span> <span class="string quoted double c"><span class="punctuation definition string begin c">&quot;</span>Hello World!<span class="punctuation definition string end c">&quot;</span></span> <span class="keyword operator arithmetic c">&lt;&lt;</span> std<span class="punctuation accessor c++">::</span>endl<span class="punctuation terminator c++">;</span>

</span></span><span class="meta function c++"><span class="meta block c++"><span class="punctuation section block end c++">}</span></span></span>

</span>
```

Long story short: Without a way to turn off the automatic addition of a newline, `parse_html_for_line` is unusable for lines with a newline character at the end, and thus for `load_defaults_newlines`.

Please note that this problem is directly connected to #317 and how that issue is addressed.

**Comment by BenjaminRi** [comment]
Fixed in `v4.5.0` with the introduction of `parse_html_for_line_which_includes_newline`. Closed.
